.github/workflows/ci.yml

```diff
@@ -65,3 +65,82 @@ jobs:
       - name: Upload to codecov
         run: |
           bash <(curl -s https://codecov.io/bash)
+
+  sentry:
+    runs-on: ubuntu-latest
+    timeout-minutes: 25
+    steps:
+      # Checkout codebase
+      - name: Checkout snuba
+        uses: actions/checkout@v2
+
+      # Checkout Sentry and run integration tests against latest snuba
+      - name: Checkout sentry
+        uses: actions/checkout@v2
+        with:
+          repository: getsentry/sentry
+          path: sentry
+
+      - name: Build snuba docker image
+        run: |
+          docker pull getsentry/snuba:latest || true
+          docker build . \
+            --build-arg PYTHON_VERSION=3.8 \
+            -t getsentry/snuba:${{ github.sha }} \
+            --cache-from getsentry/snuba:latest \
+            --target testing
+
+      # setup python
+      - uses: actions/setup-python@v1
+        with:
+          python-version: 3.8
+
+      - name: Setup steps
+        id: setup
+        run: |
+          pip install --upgrade pip wheel
+          echo "::set-output name=pip-cache-dir::$(pip cache dir)"
+          # We cannot execute actions that are not placed under .github of the main repo
+          mkdir -p .github/actions/setup-sentry/
+          cp sentry/.github/actions/setup-sentry/action.yml .github/actions/setup-sentry/action.yml
+
+      - name: Sentry's pip cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.setup.outputs.pip-cache-dir }}
+          key: sentry-deps-${{ hashFiles('sentry/requirements**.txt') }}
+          restore-keys: sentry-deps-
+
+      - name: Setup Sentry
+        id: setup-sentry
+        uses: ./.github/actions/setup-sentry
+        with:
+          workdir: sentry
+          snuba: false
+          kafka: true
+          clickhouse: true
+
+      - name: Start snuba
+        run: |
+          docker run -d --rm \
+            -p 127.0.0.1:1218:1218 \
+            -e PYTHONUNBUFFERED=1 \
+            -e SNUBA_SETTINGS=docker \
+            -e DEBUG=1 \
+            -e DEFAULT_BROKERS=sentry_kafka:9092 \
+            -e CLICKHOUSE_HOST=sentry_clickhouse \
+            -e CLICKHOUSE_PORT=9000 \
+            -e CLICKHOUSE_HTTP_PORT=8123 \
+            -e REDIS_HOST=sentry_redis \
+            -e REDIS_PORT=6379 \
+            -e REDIS_DB=1 \
+            --name sentry_snuba \
+            --network sentry \
+            getsentry/snuba:${{ github.sha }}
+          docker exec sentry_snuba snuba migrations migrate --force
```

Comment:
Can we not encapsulate these steps in a common `snuba_test` or `devservices up --test` command?